### PR TITLE
EP-7421: RequestJob API create checks if a similar job is running

### PR DESCRIPTION
**Pull request by luca-drf**

### CODE REVIEW

File: src/ensembl/production/dbcopy/api/viewsets.py [hunk 1]
```diff
@@ -27,6 +30,30 @@ class RequestJobViewSet(mixins.CreateModelMixin,
...
+            "src_host",
+            "src_incl_db",
+            "src_skip_db",
+            "src_incl_tables",
+            "src_skip_tables",
+            "tgt_host",
+            "tgt_db_name",
+        }
+        filters = {k: v for k, v in request.data.items() if k in params and v}
+        for job in RequestJob.equivalent_running_jobs(**filters):
```
Comment (marcoooo) [inline src/ensembl/production/dbcopy/api/viewsets.py:44]: See below about this comment
```suggestion
        for job in RequestJob.objects.equivalent_running_jobs(**filters):
```

File: src/ensembl/production/dbcopy/models.py [hunk 2]
```diff
@@ -180,6 +185,25 @@ def _clean_db_set_for_filters(self, from_host, field):
         except ValueError as e:
             raise ValidationError({field: str(e)})
 
+    @staticmethod
```
Comment (marcoooo) [inline src/ensembl/production/dbcopy/models.py:188]: To be inline with Django Framework, this method shouldn't be located within RequestJob Model class, but in a dedicated `Manager` one. 
```
class RequestJobManager(models.Manager):
        def equivalent_running_jobs(self, **kwargs):
           pass
```

```
class RequestJob(models.Model):
 ...
   objects = RequestJobManager()
```

Then you'll call it with

**Review by marcoooo — CHANGES_REQUESTED** [review]

File: src/ensembl/production/dbcopy/models.py (hunk 2 above)
Comment (luca-drf) [reply src/ensembl/production/dbcopy/models.py]: Good idea, thanks for the tip! See e529355

File: src/ensembl/production/dbcopy/api/viewsets.py (hunk 1 above)
Comment (luca-drf) [reply src/ensembl/production/dbcopy/api/viewsets.py]: Please see e529355

**Review by marcoooo — APPROVED** [review]

**Review by marcoooo — DISMISSED** [review]

File: src/ensembl/production/dbcopy/models.py [hunk 3]
```diff
@@ -359,6 +365,11 @@ def completion(self):
             self.progress
         )
 
+    @property
+    def equivalent_jobs(self):
+        params = {k: getattr(self, k) for k in self.__class__.objects._EQ_PARAMS.keys()}
+        return self.__class__.objects.equivalent_jobs(**params)
```
Comment (marcoooo) [inline src/ensembl/production/dbcopy/models.py:371]: No need of the __class__ call here, it's resolved with the `objects = RequestJobManager()` Which is missing?

**Review by marcoooo — CHANGES_REQUESTED** [review]
So you did change your mind about the `clean` method?

**Review by marcoooo — APPROVED** [review]

**Review by marcoooo — APPROVED** [review]

**Pull request merged**